src/System.Text.RegularExpressions/tests/Performance/Perf.Regex.Cache.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Xunit.Performance;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class Perf_Regex_Cache
+    {
+        private const int N = 50_000;
+        private const int UniqueRegsNum = (int)(N * 0.02);
+        private const int CacheSize = (int)(N * 0.02);
+        private static volatile bool s_IsMatch;
+        private readonly string[] _regexes; // shuffled
+
+        public Perf_Regex_Cache()
+        {
+            _regexes = new string[N];
+            // create:
+            {
+                var i = 0;
+                for (; i < UniqueRegsNum; i++)
+                {
+                    // "(0+)" "(1+)" ..  "(9+)(9+)(8+)" ..
+                    var re = new StringBuilder();
+                    foreach (var c in i.ToString())
+                        re.Append("(" + c + "+)");
+                    _regexes[i] = re.ToString();
+                }
+
+                for (; i < N; i++) _regexes[i] = _regexes[i % UniqueRegsNum];
+            }
+            // shuffle:
+            var random = new Random();
```

Comment:
do you want to seed for reproducibility? 42 seems favored here.

Reply:
Ok. [Looks](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Runtime.Numerics/tests/BigInteger/op_xor.cs#L12) like not everyone favors 42 :)